src/pruna/evaluation/metrics/metric_vbench_background_consistency.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2025 - Pruna AI GmbH. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Any, List
+
+import clip
+import torch
+import torch.nn.functional as F  # noqa: N812
+from torchvision.transforms.functional import convert_image_dtype
+from vbench.utils import clip_transform, init_submodules
+
+from pruna.engine.utils import set_to_best_available_device
+from pruna.evaluation.metrics.metric_stateful import StatefulMetric
+from pruna.evaluation.metrics.registry import MetricRegistry
+from pruna.evaluation.metrics.result import MetricResult
+from pruna.evaluation.metrics.utils import PAIRWISE, SINGLE, get_call_type_for_single_metric, metric_data_processor
+from pruna.evaluation.metrics.vbench_utils import VBenchMixin
+from pruna.logging.logger import pruna_logger
+
+METRIC_VBENCH_BACKGROUND_CONSISTENCY = "background_consistency"
+
+
+@MetricRegistry.register(METRIC_VBENCH_BACKGROUND_CONSISTENCY)
+class VBenchBackgroundConsistency(StatefulMetric, VBenchMixin):
+    """
+    Background Consistency metric for VBench.
+
+    Parameters
+    ----------
+    *args : Any
+        The arguments to pass to the metric.
+    device : str | None
+        The device to run the metric on.
+    call_type : str
+        The call type to use for the metric.
+    **kwargs : Any
+        The keyword arguments to pass to the metric.
+    """
+
+    metric_name: str = METRIC_VBENCH_BACKGROUND_CONSISTENCY
+    default_call_type: str = "y"  # We just need the outputs
+    higher_is_better: bool = True
+    # https://github.com/Vchitect/VBench/blob/dc62783c0fb4fd333249c0b669027fe102696682/evaluate.py#L111
+    # explicitly sets the device to cuda. We respect this here.
+    runs_on: List[str] = ["cuda"]
+    modality: List[str] = ["video"]
+    # state
+    similarity_scores: torch.Tensor
+    n_samples: torch.Tensor
+
+    def __init__(
+        self,
+        *args: Any,
+        device: str | None = None,
+        call_type: str = SINGLE,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+
+        if device is not None and str(device).split(":")[0] not in self.runs_on:
+            pruna_logger.error(f"Unsupported device {device}; supported: {self.runs_on}")
```

Comment:
perhaps we can copy assign the metric to a variable and also pass it to the ValuError?